sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkGetColumnsOperation.scala

```diff
@@ -102,8 +102,9 @@ private[hive] class SparkGetColumnsOperation(
 
       // Global temporary views
       val globalTempViewDb = catalog.globalTempViewManager.database
-      val databasePattern = Pattern.compile(CLIServiceUtils.patternToRegex(schemaName))
-      if (databasePattern.matcher(globalTempViewDb).matches()) {
+      if (schemaName == null || schemaName.isEmpty || schemaName == "*"
+        || Pattern.compile(CLIServiceUtils.patternToRegex(schemaName))
+        .matcher(globalTempViewDb).matches()) {
```

Comment:
It is by design. May be we should fix SPARK-31634 first. Another related issue: https://issues.apache.org/jira/browse/HIVE-23359

Reply:
cc @yaooqinn

Reply:
Thanks for review @wangyum. SPARK-31634 talks about sql command, but this change affect thrift meta api. Can you clarify the relation between them?

Reply:
@yaooqinn Any suggestions?